crates/rpc/rpc-eth-api/src/helpers/block.rs

```diff
@@ -233,18 +233,41 @@ pub trait LoadBlock: LoadPendingBlock + SpawnBlocking {
                 };
             }
 
+            // We retrieve the block hash for the given block id via the provider
             let block_hash = match LoadPendingBlock::provider(self)
                 .block_hash_for_id(block_id)
                 .map_err(Self::Error::from_eth_err)?
             {
                 Some(block_hash) => block_hash,
+                // If the block hash is not found, we return `None` directly
                 None => return Ok(None),
             };
 
-            self.cache()
-                .get_sealed_block_with_senders(block_hash)
-                .await
-                .map_err(Self::Error::from_eth_err)
+            // Initialize the maximum number of retries for handling reorg cases.
+            //
+            // A reorg may cause the latest block to be temporarily absent from the cache.
+            //
+            // By retrying once, we give the system a chance to update the cache with
```

Comment:
we only need to check here if the BlockId is `Latest` and only need a specialization for this tag because this only really affects the latest tag

Reply:
Just provided the retry to 1 if latest and 0 if not to solve this